### PR TITLE
Prevent dispensers from bonemealing plants when disable-bonemeal is true

### PR DESCRIPTION
**Pull request by NCBPFluffyBear**
This fix prevents dispensers from bypassing EG bonemealing when it is disabled.

### CODE REVIEW

**Comment by WalshyDev** [comment]
I feel like this should really be fixed in Spigot. BlockFertilizeEvent is made for this... I feel like a proper fix should be PR'd there or at least an issue created (not sure if one exists already)

File: src/main/java/io/github/thebusybiscuit/exoticgarden/listeners/PlantsListener.java [hunk 1]
```diff
@@ -395,6 +397,28 @@ public void onBonemealPlant(BlockFertilizeEvent e) {
         }
     }
 
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onDispenserBonemealPlant(BlockDispenseEvent e) {
+        ItemStack bonemeal = e.getItem();
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/listeners/PlantsListener.java:402]: This shouldn't really be named `bonemeal` here. We don't know if it is bone meal yet.

File: src/main/java/io/github/thebusybiscuit/exoticgarden/listeners/PlantsListener.java [hunk 2]
```diff
@@ -395,6 +397,28 @@ public void onBonemealPlant(BlockFertilizeEvent e) {
         }
     }
 
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/exoticgarden/listeners/PlantsListener.java:400]: Why HIGHEST? :NotLikeThis:

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
As Walshy already pointed out - this is such a bullshit thing to do, the `BlockFertilizeEvent` should really just be fixed within Spigot itself...

**Comment by WalshyDev** [comment]
Just looked at Jira, there is an issue for this - https://hub.spigotmc.org/jira/browse/SPIGOT-6020

Also an open PR (made by the guy who commented): https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/pull-requests/826/overview

No idea if TheCreeperCow saw this PR but that's fantastic timing

**Comment by WalshyDev** [comment]
> Just looked at Jira, there is an issue for this - https://hub.spigotmc.org/jira/browse/SPIGOT-6020
> 
> Also an open PR (made by the guy who commented): https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/pull-requests/826/overview
> 
> No idea if TheCreeperCow saw this PR but that's fantastic timing

Merged - https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/7ea8b96b4a9a168723d1eb109c5abc6fe91b78d6

**Pull request closed without merging**